internal/testrunner/runners/system/servicedeployer/compose.go

```diff
@@ -33,14 +34,20 @@ type dockerComposeDeployedService struct {
 
 	ymlPaths []string
 	project  string
+	appEnv   []string
 	sv       ServiceVariant
 }
 
 // NewDockerComposeServiceDeployer returns a new instance of a DockerComposeServiceDeployer.
 func NewDockerComposeServiceDeployer(ymlPaths []string, sv ServiceVariant) (*DockerComposeServiceDeployer, error) {
+	return newDockerComposeServiceDeployer(ymlPaths, nil, sv)
+}
+
+func newDockerComposeServiceDeployer(ymlPaths, appEnv []string, sv ServiceVariant) (*DockerComposeServiceDeployer, error) {
 	return &DockerComposeServiceDeployer{
 		ymlPaths: ymlPaths,
 		sv:       sv,
+		appEnv:   appEnv,
```

Comment:
Is there an option not to pass `appEnv` through many places?

Reply:
I couldn't find any non-intrusive one, I thought is better to pass this aroung that modifying the composer code too much.

Reply:
Yeah yeah, that's why I proposed to loosen the dependency between Compose service deployer and this one :)